routers/repo/setting.go

```diff
@@ -378,6 +378,11 @@ func SettingsPost(ctx *context.Context, form auth.RepoSettingForm) {
 		}
 
 		oldOwnerID := ctx.Repo.Owner.ID
+		// Close the GitRepo if ope
+		if ctx.Repo.GitRepo != nil {
+			ctx.Repo.GitRepo.Close()
```

Comment:
I think this is necessary in `SettingsPost()` for the rename action as well.

Reply:
Should we close the git repository on the router? Or maybe we should close it on middleware which load it after invoke the router.

Reply:
@guillep2k done now.

@lunny I thought I had done this but I realise that it must have not saved. See the changes in modules/context/repo.go